thcrap_proton

```diff
@@ -71,6 +75,40 @@ if [ ! -f "$THCRAP_FOLDER/config/$THCRAP_CONFIG" ]; then
     fi
 fi
 
+if [ "$USE_THPRAC" == 1 ]; then
+    THPRAC_VERSION=$(LD_LIBRARY_PATH=/usr/lib64:/usr/lib curl --connect-timeout 5 -si https://github.com/touhouworldcup/thprac/releases/latest | grep location: | sed 's@.*/@@' | tr -d '[:space:]')
+    if [[ -n "$THPRAC_VERSION" && ! -f "$THPRAC_FOLDER/thprac.$THPRAC_VERSION.exe" ]]; then
+        # Install latest version
+        mkdir -p "$THPRAC_FOLDER"
+        if [ ! -w "$THPRAC_FOLDER" ]; then
+            ${ZENITY} --error --text="The directory \"$THPRAC_FOLDER\" could not be created or is not writable. Check your permissions and try again."
+            exit
+        fi
+        LD_LIBRARY_PATH=/usr/lib64:/usr/lib curl -L -f "https://github.com/touhouworldcup/thprac/releases/download/$THPRAC_VERSION/thprac.$THPRAC_VERSION.exe" -o "$THPRAC_FOLDER/thprac.$THPRAC_VERSION.exe" | ${ZENITY} --progress --pulsate --auto-close --text="Downloading \"thprac.$THPRAC_VERSION.exe\""
+        if [ $? != 0 ]; then
+            # Only abort if there's no version currently installed
+            if [ ! -f "$THPRAC_FOLDER/thprac.exe" ]; then
+                ${ZENITY} --error --text="Download failed. Aborting"
+                exit
+            fi
+        else
+            # Remove old versions, only keep 5 latest versions
+            readarray -d '' THPRAC_OLD_VERSIONS < <(find "$THPRAC_FOLDER" -maxdepth 1 -regextype egrep -iregex '.*/thprac.v([0-9]|\.)*.exe' -print0 | sort -zVr | tail -z -n +6)
+            if (( ${#THPRAC_OLD_VERSIONS[@]} )); then
+                # THPRAC_OLD_VERSIONS not empty
+                rm "${THPRAC_OLD_VERSIONS[@]}"
+            fi
```

Comment:
For the sake of simplicity I'd prefer to not even bother with the older versions and just leave everything there.

Reply:
I still prefer it could automatically cleanup unused versions.

What about only keep latest version with this single command
```bash
find "$THPRAC_FOLDER" -maxdepth 1 \! -name "thprac.$THPRAC_VERSION.exe" -exec rm {} \;`
```

or replace _if_ expression with
```
(( ${#THPRAC_OLD_VERSIONS[@]} )) && rm "${THPRAC_OLD_VERSIONS[@]}"
```
it just **looks** less cluttered though.

Reply:
Keeping only the latest version seems the most solid approach. Though we better filter exactly what we want to delete, otherwise we might end up accidentally erasing user files if they set `THPRAC_FOLDER` incorrectly, e.g., to their home directory.

    find "$THPRAC_FOLDER" -maxdepth 1 -name "thprac.*.exe" \! -name "thprac.$THPRAC_VERSION.exe" -exec rm {} \;